lib/classification_lifecycle.rb

```diff
@@ -1,82 +1,106 @@
-      instance_eval(&block) if block_given?
+      update_counters
+      process_project_preference
       create_recent
       update_seen_subjects
     end
+
     publish_data
+    notify_cellect
     #to avoid duplicates in queue, do not refresh the queue before updating seen subjects
     refresh_queue
   end
 
-  def process_project_preference
-    if should_create_project_preference?
-      upp = UserProjectPreference.where(user: user, project: project).first_or_initialize do |up|
-        up.preferences = {}
-      end
-      if first_classifcation = upp.email_communication.nil?
-        ProjectClassificationsCountWorker.perform_async(project.id)
-        upp.email_communication = user.project_email_communication
-      end
-      upp.changed? ? upp.save! : upp.touch
+  def update_classification_data
+    mark_expert_classifier
+    add_seen_before_for_user
+    add_project_live_state
+    add_user_groups
+    add_lifecycled_at
+    classification.save!
+  end
+
+  def update_counters
+    return unless should_count_towards_retirement?
+
+    classification.subject_ids.each do |sid|
+      ClassificationCountWorker.perform_async(sid, classification.workflow.id, update?)
     end
   end
 
+  def process_project_preference
+    return unless should_create_project_preference?
+
+    upp = UserProjectPreference.where(user: user, project: project).first_or_initialize { |up| up.preferences = {} }
+    upp.email_communication ||= user.project_email_communication
+    upp.changed? ? upp.save! : upp.touch
+  end
+
+  def create_recent
+    return unless should_update_seen? && subjects_are_unseen_by_user?
+
+    Recent.create_from_classification(classification)
+  end
+
   def update_seen_subjects
-    if should_update_seen? && subjects_are_unseen_by_user?
-      UserSeenSubject.add_seen_subjects_for_user(**user_workflow_subject)
-      if Panoptes.use_cellect?(workflow)
-        subject_ids.each do |subject_id|
-          SeenCellectWorker.perform_async(workflow.id, user.try(:id), subject_id)
-        end
-      end
-    end
+    return unless should_update_seen? && subjects_are_unseen_by_user?
+
+    UserSeenSubject.add_seen_subjects_for_user(**user_workflow_subject)
   end
 
   def publish_data
-    if classification.complete?
-      PublishClassificationWorker.perform_async(classification.id)
+    return unless classification.complete?
+
+    PublishClassificationWorker.perform_async(classification.id)
+  end
+
+  def notify_cellect
+    return unless should_update_seen?
+    return unless subjects_are_unseen_by_user?
+    return unless Panoptes.use_cellect?(workflow)
+
+    subject_ids.each do |subject_id|
+      SeenCellectWorker.perform_async(workflow.id, user.try(:id), subject_id)
     end
   end
 
   def refresh_queue
     subjects_workflow_subject_sets.each do |set_id|
       queue = SubjectQueue.by_set(set_id).find_by(user: user, workflow: workflow)
+
```

Comment:
I think we should just remove this stuff and i'll rebase my queue removal PR later.